qiskit/optimization/algorithms/optimization_algorithm.py

```diff
@@ -92,18 +92,25 @@ def __init__(self, x: Union[List[float], np.ndarray], fval: float,
         Raises:
             QiskitOptimizationError: if sizes of ``x`` and ``variables`` do not match.
         """
-        if len(x) != len(variables):
-            raise QiskitOptimizationError(
-                'Inconsistent size of optimal value and variables. x: size {} {}, '
-                'variables: size {} {}'.format(len(x), x, len(variables),
-                                               [v.name for v in variables]))
-        self._x = x if isinstance(x, np.ndarray) else np.array(x)  # pylint: disable=invalid-name
+        self._variables = variables
+        self._variable_names = [var.name for var in self._variables]
+        if x is None:
+            # if no state is given, it is set to None
+            self._x = None  # pylint: disable=invalid-name
```

Comment:
Having x allow None here is really violating the typehint for x in the constructor and in the getter for x that are not declared Optional. I guess _x could be set to an empty array and variables dict to an empty dict {}, rather than None, to address that aspect in the getters. As to the constructor, if this is the way to handle it, then it should be Optional and maybe the empty array x and variables_dict should be specified in the docstring as the result of doing this.

Reply:
Good point, I'll update the type hint and doc strings. I think `None` is cleaner. An empty list / dict just means that you have to catch the problem one step later. If you are in a setting where an algorithm can fail, the best would be to check for the result status in the first place.